### ISSUE TITLE
Unexpected shutdowns with postgres

### ISSUE THREAD

**Issue by Esshahn** [ether/etherpad-lite#5130]
**Describe the bug**
Etherpad is shutting down after fatal errors, resulting in system reboots (eg. 4 times today)

**To Reproduce**
1. etherpad lite deployed on render.com
2. postgres deployed on render.com

Console log:
```
Jul 26 12:49:58 PM  [2021-07-26 10:49:58.354] [INFO] Minify - Compress JS file /opt/etherpad-lite/src/static/js/pad_userlist.js.
Jul 26 12:49:58 PM  [2021-07-26 10:49:58.462] [INFO] Minify - Compress JS file /opt/etherpad-lite/src/static/js/vendors/farbtastic.js.
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.265] [ERROR] server - Metrics at time of fatal error:
Jul 26 12:50:00 PM  {
Jul 26 12:50:00 PM    "httpStartTime": 1627295664459,
Jul 26 12:50:00 PM    "memoryUsage": 121081856,
Jul 26 12:50:00 PM    "memoryUsageHeap": 32899952,
Jul 26 12:50:00 PM    "ueberdb_lockAwaits": 0,
Jul 26 12:50:00 PM    "ueberdb_lockAcquires": 1,
Jul 26 12:50:00 PM    "ueberdb_lockReleases": 1,
Jul 26 12:50:00 PM    "ueberdb_reads": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFailed": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFinished": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFromCache": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFromDb": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFromDbFailed": 0,
Jul 26 12:50:00 PM    "ueberdb_readsFromDbFinished": 0,
Jul 26 12:50:00 PM    "ueberdb_writes": 1,
Jul 26 12:50:00 PM    "ueberdb_writesFailed": 0,
Jul 26 12:50:00 PM    "ueberdb_writesFinished": 0,
Jul 26 12:50:00 PM    "ueberdb_writesObsoleted": 0,
Jul 26 12:50:00 PM    "ueberdb_writesToDb": 1,
Jul 26 12:50:00 PM    "ueberdb_writesToDbFailed": 0,
Jul 26 12:50:00 PM    "ueberdb_writesToDbFinished": 0,
Jul 26 12:50:00 PM    "ueberdb_writesToDbRetried": 0,
Jul 26 12:50:00 PM    "totalUsers": 0,
Jul 26 12:50:00 PM    "activePads": 0,
Jul 26 12:50:00 PM    "httpRequests": {
Jul 26 12:50:00 PM      "meter": {
Jul 26 12:50:00 PM        "mean": 0,
Jul 26 12:50:00 PM        "count": 0,
Jul 26 12:50:00 PM        "currentRate": 0,
Jul 26 12:50:00 PM        "1MinuteRate": 0,
Jul 26 12:50:00 PM        "5MinuteRate": 0,
Jul 26 12:50:00 PM        "15MinuteRate": 0
Jul 26 12:50:00 PM      },
Jul 26 12:50:00 PM      "histogram": {
Jul 26 12:50:00 PM        "min": null,
Jul 26 12:50:00 PM        "max": null,
Jul 26 12:50:00 PM        "sum": 0,
Jul 26 12:50:00 PM        "variance": null,
Jul 26 12:50:00 PM        "mean": 0,
Jul 26 12:50:00 PM        "stddev": null,
Jul 26 12:50:00 PM        "count": 0,
Jul 26 12:50:00 PM        "median": null,
Jul 26 12:50:00 PM        "p75": null,
Jul 26 12:50:00 PM        "p95": null,
Jul 26 12:50:00 PM        "p99": null,
Jul 26 12:50:00 PM        "p999": null
Jul 26 12:50:00 PM      }
Jul 26 12:50:00 PM    }
Jul 26 12:50:00 PM  }
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.266] [ERROR] server - Error: Connection terminated unexpectedly    at Connection.<anonymous> (/opt/etherpad-lite/src/node_modules/pg/lib/client.js:132:73)    at Object.onceWrapper (events.js:481:28)    at Connection.emit (events.js:375:28)    at Connection.emit (domain.js:470:12)    at Socket.<anonymous> (/opt/etherpad-lite/src/node_modules/pg/lib/connection.js:57:12)    at Socket.emit (events.js:375:28)    at Socket.emit (domain.js:470:12)    at TCP.<anonymous> (net.js:675:12)
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.266] [INFO] server - Exiting...
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.266] [INFO] server - Stopping Etherpad...
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.267] [INFO] http - Closing HTTP server...
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.267] [INFO] socket.io - Closing socket.io engine...
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.268] [INFO] socket.io - All socket.io clients have disconnected
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.274] [INFO] http - Waiting for 1 HTTP clients to disconnect...
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.279] [INFO] http - HTTP server closed
Jul 26 12:50:00 PM  [2021-07-26 10:50:00.279] [ERROR] console - Error: Connection terminated unexpectedly    at Connection.<anonymous> (/opt/etherpad-lite/src/node_modules/pg/lib/client.js:132:73)    at Object.onceWrapper (events.js:481:28)    at Connection.emit (events.js:375:28)    at Connection.emit (domain.js:470:12)    at Socket.<anonymous> (/opt/etherpad-lite/src/node_modules/pg/lib/connection.js:57:12)    at Socket.emit (events.js:375:28)    at Socket.emit (domain.js:470:12)    at TCP.<anonymous> (net.js:675:12)
Jul 26 12:50:03 PM  [2021-07-26 10:50:03.271] [ERROR] server - Error occurred while stopping Etherpad
Jul 26 12:50:03 PM  [2021-07-26 10:50:03.272] [ERROR] server - Metrics at time of fatal error:
Jul 26 12:50:03 PM  {
Jul 26 12:50:03 PM    "httpStartTime": 0,
Jul 26 12:50:03 PM    "memoryUsage": 80666624,
Jul 26 12:50:03 PM    "memoryUsageHeap": 33113736,
Jul 26 12:50:03 PM    "ueberdb_lockAwaits": 0,
Jul 26 12:50:03 PM    "ueberdb_lockAcquires": 1,
Jul 26 12:50:03 PM    "ueberdb_lockReleases": 1,
Jul 26 12:50:03 PM    "ueberdb_reads": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFailed": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFinished": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFromCache": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFromDb": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFromDbFailed": 0,
Jul 26 12:50:03 PM    "ueberdb_readsFromDbFinished": 0,
Jul 26 12:50:03 PM    "ueberdb_writes": 1,
Jul 26 12:50:03 PM    "ueberdb_writesFailed": 1,
Jul 26 12:50:03 PM    "ueberdb_writesFinished": 1,
Jul 26 12:50:03 PM    "ueberdb_writesObsoleted": 0,
Jul 26 12:50:03 PM    "ueberdb_writesToDb": 1,
Jul 26 12:50:03 PM    "ueberdb_writesToDbFailed": 1,
Jul 26 12:50:03 PM    "ueberdb_writesToDbFinished": 1,
Jul 26 12:50:03 PM    "ueberdb_writesToDbRetried": 0,
Jul 26 12:50:03 PM    "totalUsers": 0,
Jul 26 12:50:03 PM    "activePads": 0,
Jul 26 12:50:03 PM    "httpRequests": {
Jul 26 12:50:03 PM      "meter": {
Jul 26 12:50:03 PM        "mean": 0,
Jul 26 12:50:03 PM        "count": 0,
Jul 26 12:50:03 PM        "currentRate": 0,
Jul 26 12:50:03 PM        "1MinuteRate": 0,
Jul 26 12:50:03 PM        "5MinuteRate": 0,
Jul 26 12:50:03 PM        "15MinuteRate": 0
Jul 26 12:50:03 PM      },
Jul 26 12:50:03 PM      "histogram": {
Jul 26 12:50:03 PM        "min": null,
Jul 26 12:50:03 PM        "max": null,
Jul 26 12:50:03 PM        "sum": 0,
Jul 26 12:50:03 PM        "variance": null,
Jul 26 12:50:03 PM        "mean": 0,
Jul 26 12:50:03 PM        "stddev": null,
Jul 26 12:50:03 PM        "count": 0,
Jul 26 12:50:03 PM        "median": null,
Jul 26 12:50:03 PM        "p75": null,
Jul 26 12:50:03 PM        "p95": null,
Jul 26 12:50:03 PM        "p99": null,
Jul 26 12:50:03 PM        "p999": null
Jul 26 12:50:03 PM      }
Jul 26 12:50:03 PM    }
Jul 26 12:50:03 PM  }
Jul 26 12:50:03 PM  [2021-07-26 10:50:03.272] [ERROR] server - Error: Timed out waiting for shutdown tasks    at Timeout._onTimeout (/opt/etherpad-lite/src/node/server.js:197:43)    at listOnTimeout (internal/timers.js:557:17)    at processTimers (internal/timers.js:500:7)
Jul 26 12:50:03 PM  [2021-07-26 10:50:03.272] [ERROR] server - Error occurred while waiting to exit. Forcing an immediate unclean exit...
Jul 26 12:50:05 PM  [2021-07-26 10:50:05.945] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: /opt/etherpad-lite
```

Settings:

```
/**
 * THIS IS THE SETTINGS FILE THAT IS COPIED INSIDE THE DOCKER CONTAINER.
 *
 * By default, some runtime customizations are supported (see the
 * documentation).
 *
 * If you need more control, edit this file and rebuild the container.
 */

/*
 * This file must be valid JSON. But comments are allowed
 *
 * Please edit settings.json, not settings.json.template
 *
 * Please note that starting from Etherpad 1.6.0 you can store DB credentials in
 * a separate file (credentials.json).
 *
 *
 * ENVIRONMENT VARIABLE SUBSTITUTION
 * =================================
 *
 * All the configuration values can be read from environment variables using the
 * syntax "${ENV_VAR}" or "${ENV_VAR:default_value}".
 *
 * This is useful, for example, when running in a Docker container.
 *
 * DETAILED RULES:
 *   - If the environment variable is set to the string "true" or "false", the
 *     value becomes Boolean true or false.
 *   - If the environment variable is set to the string "null", the value
 *     becomes null.
 *   - If the environment variable is set to the string "undefined", the setting
 *     is removed entirely, except when used as the member of an array in which
 *     case it becomes null.
 *   - If the environment variable is set to a string representation of a finite
 *     number, the string is converted to that number.
 *   - If the environment variable is set to any other string, including the
 *     empty string, the value is that string.
 *   - If the environment variable is unset and a default value is provided, the
 *     value is as if the environment variable was set to the provided default:
 *       - "${UNSET_VAR:}" becomes the empty string.
 *       - "${UNSET_VAR:foo}" becomes the string "foo".
 *       - "${UNSET_VAR:true}" and "${UNSET_VAR:false}" become true and false.
 *       - "${UNSET_VAR:null}" becomes null.
 *       - "${UNSET_VAR:undefined}" causes the setting to be removed (or be set
 *         to null, if used as a member of an array).
 *   - If the environment variable is unset and no default value is provided,
 *     the value becomes null. THIS BEHAVIOR MAY CHANGE IN A FUTURE VERSION OF
 *     ETHERPAD; if you want the default value to be null, you should explicitly
 *     specify "null" as the default value.
 *
 * EXAMPLE:
 *    "port":     "${PORT:9001}"
 *    "minify":   "${MINIFY}"
 *    "skinName": "${SKIN_NAME:colibris}"
 *
 * Would read the configuration values for those items from the environment
 * variables PORT, MINIFY and SKIN_NAME.
 *
 * If PORT and SKIN_NAME variables were not defined, the default values 9001 and
 * "colibris" would be used.
 * The configuration value "minify", on the other hand, does not have a
 * designated default value. Thus, if the environment variable MINIFY were
 * undefined, "minify" would be null.
 *
 * REMARKS:
 * 1) please note that variable substitution always needs to be quoted.
 *
 *    "port":     9001,            <-- Literal values. When not using
 *    "minify":   false                substitution, only strings must be
 *    "skinName": "colibris"           quoted. Booleans and numbers must not.
 *
 *    "port":     "${PORT:9001}"   <-- CORRECT: if you want to use a variable
 *    "minify":   "${MINIFY:true}"     substitution, put quotes around its name,
 *    "skinName": "${SKIN_NAME}"       even if the required value is a number or
 *                                     a boolean.
 *                                     Etherpad will take care of rewriting it
 *                                     to the proper type if necessary.
 *
 *    "port":     ${PORT:9001}     <-- ERROR: this is not valid json. Quotes
 *    "minify":   ${MINIFY}            around variable names are missing.
 *    "skinName": ${SKIN_NAME}
 *
 * 2) Beware of undefined variables and default values: nulls and empty strings
 *    are different!
 *
 *    This is particularly important for user's passwords (see the relevant
 *    section):
 *
 *    "password": "${PASSW}"  // if PASSW is not defined would result in password === null
 *    "password": "${PASSW:}" // if PASSW is not defined would result in password === ''
 *
 *    If you want to use an empty value (null) as default value for a variable,
 *    simply do not set it, without putting any colons: "${ABIWORD}".
 *
 * 3) if you want to use newlines in the default value of a string parameter,
 *    use "\n" as usual.
 *
 *    "defaultPadText" : "${DEFAULT_PAD_TEXT}Line 1\nLine 2"
 */
{
  /*
   * Name your instance!
   */
  "title": "${TITLE:Digital Vereint Etherpad}",

  /*
   * Pathname of the favicon you want to use. If null, the skin's favicon is
   * used if one is provided by the skin, otherwise the default Etherpad favicon
   * is used. If this is a relative path it is interpreted as relative to the
   * Etherpad root directory.
   */
  "favicon": "${FAVICON:null}",

  /*
   * Skin name.
   *
   * Its value has to be an existing directory under src/static/skins.
   * You can write your own, or use one of the included ones:
   *
   * - "no-skin":  an empty skin (default). This yields the unmodified,
   *               traditional Etherpad theme.
   * - "colibris": the new experimental skin (since Etherpad 1.8), candidate to
   *               become the default in Etherpad 2.0
   */
  "skinName": "${SKIN_NAME:colibris}",

  /*
   * Skin Variants
   *
   * Use the UI skin variants builder at /p/test#skinvariantsbuilder
   *
   * For the colibris skin only, you can choose how to render the three main
   * containers:
   * - toolbar (top menu with icons)
   * - editor (containing the text of the pad)
   * - background (area outside of editor, mostly visible when using page style)
   *
   * For each of the 3 containers you can choose 4 color combinations:
   * super-light, light, dark, super-dark.
   *
   * For example, to make the toolbar dark, you will include "dark-toolbar" into
   * skinVariants.
   *
   * You can provide multiple skin variants separated by spaces. Default
   * skinVariant is "super-light-toolbar super-light-editor light-background".
   *
   * For the editor container, you can also make it full width by adding
   * "full-width-editor" variant (by default editor is rendered as a page, with
   * a max-width of 900px).
   */
  "skinVariants": "${SKIN_VARIANTS:super-dark-toolbar super-light-editor dark-background}",

  /*
   * IP and port which Etherpad should bind at.
   *
   * Binding to a Unix socket is also supported: just use an empty string for
   * the ip, and put the full path to the socket in the port parameter.
   *
   * EXAMPLE USING UNIX SOCKET:
   *    "ip": "",                             // <-- has to be an empty string
   *    "port" : "/somepath/etherpad.socket", // <-- path to a Unix socket
   */
  "ip": "${IP:0.0.0.0}",
  "port": "${PORT:9001}",

  /*
   * Option to hide/show the settings.json in admin page.
   *
   * Default option is set to true
   */
  "showSettingsInAdminPage": "${SHOW_SETTINGS_IN_ADMIN_PAGE:true}",

  /*
   * Node native SSL support
   *
   * This is disabled by default.
   * Make sure to have the minimum and correct file access permissions set so
   * that the Etherpad server can access them
   */

  /*
  "ssl" : {
            "key"  : "/path-to-your/epl-server.key",
            "cert" : "/path-to-your/epl-server.crt",
            "ca": ["/path-to-your/epl-intermediate-cert1.crt", "/path-to-your/epl-intermediate-cert2.crt"]
          },
  */

  /*
   * The type of the database.
   *
   * You can choose between many DB drivers, for example: dirty, postgres,
   * sqlite, mysql.
   *
   * You shouldn't use "dirty" for for anything else than testing or
   * development.
   *
   *
   * Database specific settings are dependent on dbType, and go in dbSettings.
   * Remember that since Etherpad 1.6.0 you can also store this information in
   * credentials.json.
   *
   * For a complete list of the supported drivers, please refer to:
   * https://www.npmjs.com/package/ueberdb2
   */

  "dbType": "${DB_TYPE}",
  "dbSettings": {
    "host":     "${DB_HOST}",
    "port":     "${DB_PORT}",
    "database": "${DB_NAME}",
    "user":     "${DB_USER}",
    "password": "${DB_PASS}",
    "charset":  "${DB_CHARSET:undefined}",
    "filename": "${DB_FILENAME:var/dirty.db}"
  },

  /*
   * The default text of a pad
   */
  "defaultPadText" : "${DEFAULT_PAD_TEXT:Willkommen beim Digital Vereint Etherpad!\n\nAlles, was Du hier schreibst, wird automatisch gespeichert. Andere Nutzer:innen mit dem Link zu dieser Seite können so in Echtzeit mit Dir zusammen arbeiten. Bitte beachte, dass Deine Informationen öffentlich für alle zugänglich sind, die den Link kennen, daher solltest Du auf private Informationen, etwa Passwörter, verzichten.\nWeitere Informationen zu Etherpad findest Du hier: https:\/\/etherpad.org\n\nDiesen Text kann Du einfach markieren und löschen.\nViel Spaß, Dein Team von Digital Vereint.\n}",

  /*
   * Default Pad behavior.
   *
   * Change them if you want to override.
   */
  "padOptions": {
    "noColors":         "${PAD_OPTIONS_NO_COLORS:false}",
    "showControls":     "${PAD_OPTIONS_SHOW_CONTROLS:true}",
    "showChat":         "${PAD_OPTIONS_SHOW_CHAT:true}",
    "showLineNumbers":  "${PAD_OPTIONS_SHOW_LINE_NUMBERS:true}",
    "useMonospaceFont": "${PAD_OPTIONS_USE_MONOSPACE_FONT:false}",
    "userName":         "${PAD_OPTIONS_USER_NAME:false}",
    "userColor":        "${PAD_OPTIONS_USER_COLOR:false}",
    "rtl":              "${PAD_OPTIONS_RTL:false}",
    "alwaysShowChat":   "${PAD_OPTIONS_ALWAYS_SHOW_CHAT:false}",
    "chatAndUsers":     "${PAD_OPTIONS_CHAT_AND_USERS:false}",
    "lang":             "${PAD_OPTIONS_LANG:en-gb}"
  },

  /*
   * Pad Shortcut Keys
   */
  "padShortcutEnabled" : {
    "altF9":     "${PAD_SHORTCUTS_ENABLED_ALT_F9:true}",      /* focus on the File Menu and/or editbar */
    "altC":      "${PAD_SHORTCUTS_ENABLED_ALT_C:true}",       /* focus on the Chat window */
    "cmdShift2": "${PAD_SHORTCUTS_ENABLED_CMD_SHIFT_2:true}", /* shows a gritter popup showing a line author */
    "delete":    "${PAD_SHORTCUTS_ENABLED_DELETE:true}",
    "return":    "${PAD_SHORTCUTS_ENABLED_RETURN:true}",
    "esc":       "${PAD_SHORTCUTS_ENABLED_ESC:true}",         /* in mozilla versions 14-19 avoid reconnecting pad */
    "cmdS":      "${PAD_SHORTCUTS_ENABLED_CMD_S:true}",       /* save a revision */
    "tab":       "${PAD_SHORTCUTS_ENABLED_TAB:true}",         /* indent */
    "cmdZ":      "${PAD_SHORTCUTS_ENABLED_CMD_Z:true}",       /* undo/redo */
    "cmdY":      "${PAD_SHORTCUTS_ENABLED_CMD_Y:true}",       /* redo */
    "cmdI":      "${PAD_SHORTCUTS_ENABLED_CMD_I:true}",       /* italic */
    "cmdB":      "${PAD_SHORTCUTS_ENABLED_CMD_B:true}",       /* bold */
    "cmdU":      "${PAD_SHORTCUTS_ENABLED_CMD_U:true}",       /* underline */
    "cmd5":      "${PAD_SHORTCUTS_ENABLED_CMD_5:true}",       /* strike through */
    "cmdShiftL": "${PAD_SHORTCUTS_ENABLED_CMD_SHIFT_L:true}", /* unordered list */
    "cmdShiftN": "${PAD_SHORTCUTS_ENABLED_CMD_SHIFT_N:true}", /* ordered list */
    "cmdShift1": "${PAD_SHORTCUTS_ENABLED_CMD_SHIFT_1:true}", /* ordered list */
    "cmdShiftC": "${PAD_SHORTCUTS_ENABLED_CMD_SHIFT_C:true}", /* clear authorship */
    "cmdH":      "${PAD_SHORTCUTS_ENABLED_CMD_H:true}",       /* backspace */
    "ctrlHome":  "${PAD_SHORTCUTS_ENABLED_CTRL_HOME:true}",   /* scroll to top of pad */
    "pageUp":    "${PAD_SHORTCUTS_ENABLED_PAGE_UP:true}",
    "pageDown":  "${PAD_SHORTCUTS_ENABLED_PAGE_DOWN:true}"
  },

  /*
   * Should we suppress errors from being visible in the default Pad Text?
   */
  "suppressErrorsInPadText": "${SUPPRESS_ERRORS_IN_PAD_TEXT:true}",

  /*
   * If this option is enabled, a user must have a session to access pads.
   * This effectively allows only group pads to be accessed.
   */
  "requireSession": "${REQUIRE_SESSION:false}",

  /*
   * Users may edit pads but not create new ones.
   *
   * Pad creation is only via the API.
   * This applies both to group pads and regular pads.
   */
  "editOnly": "${EDIT_ONLY:false}",

  /*
   * If true, all css & js will be minified before sending to the client.
   *
   * This will improve the loading performance massively, but makes it difficult
   * to debug the javascript/css
   */
  "minify": "${MINIFY:true}",

  /*
   * How long may clients use served javascript code (in seconds)?
   *
   * Not setting this may cause problems during deployment.
   * Set to 0 to disable caching.
   */
  "maxAge": "${MAX_AGE:21600}", // 60 * 60 * 6 = 6 hours

  /*
   * Absolute path to the Abiword executable.
   *
   * Abiword is needed to get advanced import/export features of pads. Setting
   * it to null disables Abiword and will only allow plain text and HTML
   * import/exports.
   */
  "abiword": "${ABIWORD:null}",

  /*
   * This is the absolute path to the soffice executable.
   *
   * LibreOffice can be used in lieu of Abiword to export pads.
   * Setting it to null disables LibreOffice exporting.
   */
  "soffice": "${SOFFICE:null}",

  /*
   * Path to the Tidy executable.
   *
   * Tidy is used to improve the quality of exported pads.
   * Setting it to null disables Tidy.
   */
  "tidyHtml": "${TIDY_HTML:null}",

  /*
   * Allow import of file types other than the supported ones:
   * txt, doc, docx, rtf, odt, html & htm
   */
  "allowUnknownFileEnds": "${ALLOW_UNKNOWN_FILE_ENDS:true}",

  /*
   * This setting is used if you require authentication of all users.
   *
   * Note: "/admin" always requires authentication.
   */
  "requireAuthentication": "${REQUIRE_AUTHENTICATION:false}",

  /*
   * Require authorization by a module, or a user with is_admin set, see below.
   */
  "requireAuthorization": "${REQUIRE_AUTHORIZATION:false}",

  /*
   * When you use NGINX or another proxy/load-balancer set this to true.
   *
   * This is especially necessary when the reverse proxy performs SSL
   * termination, otherwise the cookies will not have the "secure" flag.
   *
   * The other effect will be that the logs will contain the real client's IP,
   * instead of the reverse proxy's IP.
   */
  "trustProxy": "${TRUST_PROXY:false}",

  /*
   * Settings controlling the session cookie issued by Etherpad.
   */
  "cookie": {
    /*
     * Value of the SameSite cookie property. "Lax" is recommended unless
     * Etherpad will be embedded in an iframe from another site, in which case
     * this must be set to "None". Note: "None" will not work (the browser will
     * not send the cookie to Etherpad) unless https is used to access Etherpad
     * (either directly or via a reverse proxy with "trustProxy" set to true).
     *
     * "Strict" is not recommended because it has few security benefits but
     * significant usability drawbacks vs. "Lax". See
     * https://stackoverflow.com/q/41841880 for discussion.
     */
    "sameSite": "${COOKIE_SAME_SITE:Lax}"
  },

  /*
   * Privacy: disable IP logging
   */
  "disableIPlogging": "${DISABLE_IP_LOGGING:true}",

  /*
   * Time (in seconds) to automatically reconnect pad when a "Force reconnect"
   * message is shown to user.
   *
   * Set to 0 to disable automatic reconnection.
   */
  "automaticReconnectionTimeout": "${AUTOMATIC_RECONNECTION_TIMEOUT:0}",

  /*
   * By default, when caret is moved out of viewport, it scrolls the minimum
   * height needed to make this line visible.
   */
  "scrollWhenFocusLineIsOutOfViewport": {

    /*
     * Percentage of viewport height to be additionally scrolled.
     *
     * E.g.: use "percentage.editionAboveViewport": 0.5, to place caret line in
     *       the middle of viewport, when user edits a line above of the
     *       viewport
     *
     * Set to 0 to disable extra scrolling
     */
    "percentage": {
      "editionAboveViewport": "${FOCUS_LINE_PERCENTAGE_ABOVE:0}",
      "editionBelowViewport": "${FOCUS_LINE_PERCENTAGE_BELOW:0}"
    },

    /*
     * Time (in milliseconds) used to animate the scroll transition.
     * Set to 0 to disable animation
     */
    "duration": "${FOCUS_LINE_DURATION:0}",

    /*
     * Flag to control if it should scroll when user places the caret in the
     * last line of the viewport
     */
    "scrollWhenCaretIsInTheLastLineOfViewport": "${FOCUS_LINE_CARET_SCROLL:false}",

    /*
     * Percentage of viewport height to be additionally scrolled when user
     * presses arrow up in the line of the top of the viewport.
     *
     * Set to 0 to let the scroll to be handled as default by Etherpad
     */
    "percentageToScrollWhenUserPressesArrowUp": "${FOCUS_LINE_PERCENTAGE_ARROW_UP:0}"
  },

  /*
   * User accounts. These accounts are used by:
   *   - default HTTP basic authentication if no plugin handles authentication
   *   - some but not all authentication plugins
   *   - some but not all authorization plugins
   *
   * User properties:
   *   - password: The user's password. Some authentication plugins will ignore
   *     this.
   *   - is_admin: true gives access to /admin. Defaults to false. If you do not
   *     uncomment this, /admin will not be available!
   *   - readOnly: If true, this user will not be able to create new pads or
   *     modify existing pads. Defaults to false.
   *   - canCreate: If this is true and readOnly is false, this user can create
   *     new pads. Defaults to true.
   *
   * Authentication and authorization plugins may define additional properties.
   *
   * WARNING: passwords should not be stored in plaintext in this file.
   *          If you want to mitigate this, please install ep_hash_auth and
   *          follow the section "secure your installation" in README.md
   */

  "users": {
    "admin": {
      // 1) "password" can be replaced with "hash" if you install ep_hash_auth
      // 2) please note that if password is null, the user will not be created
      "password": "${ADMIN_PASSWORD:null}",
      "is_admin": true
    },
    "user": {
      // 1) "password" can be replaced with "hash" if you install ep_hash_auth
      // 2) please note that if password is null, the user will not be created
      "password": "${USER_PASSWORD:null}",
      "is_admin": false
    }
  },

  /*
   * Restrict socket.io transport methods
   */
  "socketTransportProtocols" : ["xhr-polling", "jsonp-polling", "htmlfile"],

  "socketIo": {
    /*
     * Maximum permitted client message size (in bytes). All messages from
     * clients that are larger than this will be rejected. Large values make it
     * possible to paste large amounts of text, and plugins may require a larger
     * value to work properly, but increasing the value increases susceptibility
     * to denial of service attacks (malicious clients can exhaust memory).
     */
    "maxHttpBufferSize": 10000
  },

  /*
   * Allow Load Testing tools to hit the Etherpad Instance.
   *
   * WARNING: this will disable security on the instance.
   */
  "loadTest": "${LOAD_TEST:false}",

  /**
  * Disable dump of objects preventing a clean exit
  */
  "dumpOnUncleanExit": false,

  /*
   * Disable indentation on new line when previous line ends with some special
   * chars (':', '[', '(', '{')
   */

  /*
  "indentationOnNewLine": false,
  */

  /*
   * From Etherpad 1.8.3 onwards, import and export of pads is always rate
   * limited.
   *
   * The default is to allow at most 10 requests per IP in a 90 seconds window.
   * After that the import/export request is rejected.
   *
   * See https://github.com/nfriedly/express-rate-limit for more options
   */
  "importExportRateLimiting": {
    // duration of the rate limit window (milliseconds)
    "windowMs": "${IMPORT_EXPORT_RATE_LIMIT_WINDOW:90000}",

    // maximum number of requests per IP to allow during the rate limit window
    "max": "${IMPORT_EXPORT_MAX_REQ_PER_IP:10}"
  },

  /*
   * From Etherpad 1.8.3 onwards, the maximum allowed size for a single imported
   * file is always bounded.
   *
   * File size is specified in bytes. Default is 50 MB.
   */
  "importMaxFileSize": "${IMPORT_MAX_FILE_SIZE:52428800}", // 50 * 1024 * 1024

  /*
   * From Etherpad 1.8.5 onwards, when Etherpad is in production mode commits from individual users are rate limited
   *
   * The default is to allow at most 10 changes per IP in a 1 second window.
   * After that the change is rejected.
   *
   * See https://github.com/animir/node-rate-limiter-flexible/wiki/Overall-example#websocket-single-connection-prevent-flooding for more options
   */
  "commitRateLimiting": {
    // duration of the rate limit window (seconds)
    "duration": "${COMMIT_RATE_LIMIT_DURATION:1}",

    // maximum number of changes per IP to allow during the rate limit window
    "points": "${COMMIT_RATE_LIMIT_POINTS:10}"
  },

  /*
   * Toolbar buttons configuration.
   *
   * Uncomment to customize.
   */

  /*
  "toolbar": {
    "left": [
      ["bold", "italic", "underline", "strikethrough"],
      ["orderedlist", "unorderedlist", "indent", "outdent"],
      ["undo", "redo"],
      ["clearauthorship"]
    ],
    "right": [
      ["importexport", "timeslider", "savedrevision"],
      ["settings", "embed"],
      ["showusers"]
    ],
    "timeslider": [
      ["timeslider_export", "timeslider_returnToPad"]
    ]
  },
  */

  /*
   * Expose Etherpad version in the web interface and in the Server http header.
   *
   * Do not enable on production machines.
   */
  "exposeVersion": "${EXPOSE_VERSION:false}",

  /*
   * The log level we are using.
   *
   * Valid values: DEBUG, INFO, WARN, ERROR
   */
  "loglevel": "${LOGLEVEL:INFO}",

  /*
   * Logging configuration. See log4js documentation for further information:
   * https://github.com/nomiddlename/log4js-node
   *
   * You can add as many appenders as you want here.
   */
  "logconfig" :
    { "appenders": [
        { "type": "console"
        //, "category": "access"// only logs pad access
        }

      /*
      , { "type": "file"
      , "filename": "your-log-file-here.log"
      , "maxLogSize": 1024
      , "backups": 3 // how many log files there're gonna be at max
      //, "category": "test" // only log a specific category
        }
      */

      /*
      , { "type": "logLevelFilter"
        , "level": "warn" // filters out all log messages that have a lower level than "error"
        , "appender":
          {  Use whatever appender you want here  }
        }
      */

      /*
      , { "type": "logLevelFilter"
        , "level": "error" // filters out all log messages that have a lower level than "error"
        , "appender":
          { "type": "smtp"
          , "subject": "An error occurred in your EPL instance!"
          , "recipients": "bar@blurdybloop.com, baz@blurdybloop.com"
          , "sendInterval": 300 // 60 * 5 = 5 minutes -- will buffer log messages; set to 0 to send a mail for every message
          , "transport": "SMTP", "SMTP": { // see https://github.com/andris9/Nodemailer#possible-transport-methods
              "host": "smtp.example.com", "port": 465,
              "secureConnection": true,
              "auth": {
                  "user": "foo@example.com",
                  "pass": "bar_foo"
              }
            }
          }
        }
      */

      ]
    }, // logconfig

  /* Override any strings found in locale directories */
  "customLocaleStrings": {}
}
````

**Comment by rhansen** [comment]
What are your `DB_*` variables set to?

**Comment by Esshahn** [comment]
I've used the appropriate settings for render.com (at least I think so, and it's worth noting that while the service is up, it's actually connecting to the DB and working fine)

````
DB_HOST: dpg-c3u1s2c7o9q2oh59tjog
DB_NAME: etherpad
DB_PASS: my password
DB_PORT: 5432
DB_TYPE: postgres
DB_USER: dvpostgres
````

**Comment by Esshahn** [comment]
I've changes `DB_TYPE` to `postgrespool`, which, so far, does seem to run stable for a couple of hours.

**Comment by rhansen** [comment]
Fixed in PR #5163.
